File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
import math
import re
def solution(n, k):
    answer = 0
    
    # n을 k진법으로 변환
    def convert(n, base):
        T = "0123456789ABCDEF"
        q, r = divmod(n,base)
        if q ==0:
            return T[r]
        else:
            return convert(q, base) + T[r]
    
    # 소수판별
    def isPrime(n):
        for i in range(2, int(math.sqrt(n))+1):
            if n % i == 0:
                return False
        return True
    
    num = convert(n,k)
    num_list = num.split('0')

    for x in num_list:
        if x and x!='1' and isPrime(int(x)):
            answer +=1
    
    return answer

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function isPrimeNumber(number) {
  if (number <= 1) {
    return false;
  }
  for (let i = 2; i <= Math.sqrt(number); i++) {
    if (number % i === 0) {
      return false;
    }
  }
  return true;
}

function solution(n, k) {
  let answer = 0;
  let num = n.toString(k); 
  let numArr = num.split("0"); // 변환한 숫자를 0을 기준으로 나눠 배열에 저장

  for (let i = 0; i < numArr.length; i++) {
    if (isPrimeNumber(Number(numArr[i]))) {
      answer++; 
    }
  }

  return answer;
}